### PR TITLE
feat ✨ : Add support for glossary bulk upload and download.

### LINKED ISSUE #9100: UI: Bulk upload of Glossary Terms - P0 - 0.13.0

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
Closes #9100 

### Describe your changes :

I worked on issue #9100 

### Export
 - Export option should be visible for the glossary
 - On click of export user should redirect to the export action page
 - The user should be able to change the file name
 - On click of the export file should be downloaded to the user system
    
### Import
- Import Option should be visible for the glossary
- On click of import user should redirect to the import action page
- The user should be able to import via either dragging the file or clicking on the browse file button
- Show the status screen after the file is uploaded
    - **Success** and **PartialSuccess**: show the preview button and on click of it show the preview and import button ( on click import the glossary terms and redirect to the glossary page )
    - **Failure**: Same as a success but hide the import button
    - **Abort**: Show the aborted reason and cancel button 

#
### Type of change :

- [x] New feature

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/59080942/212669308-8daa10a0-3c4d-48e9-9c77-028b2c1550fc.mov

https://user-images.githubusercontent.com/59080942/212669344-c4c12e99-c06e-465a-8551-e10f9d5b3aa0.mov

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryV1.component.tsx [hunk 1]
```diff
@@ -266,6 +276,68 @@ const GlossaryV1 = ({
...
+                    <Col span={21}>
+                      <Typography.Text
+                        className="font-medium"
+                        data-testid="export-button-title">
+                        {t('label.export')}
+                      </Typography.Text>
+                    </Col>
+                    <Col className="p-t-xss">
+                      <Typography.Paragraph className="text-grey-muted text-xs m-b-0 line-height-16">
+                        Export glossary terms
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryV1.component.tsx:304]: localisation

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/ExportGlossaryModal/ExportGlossaryModal.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import { showErrorToast } from 'utils/ToastUtils';
+
+interface Props {
+  isModalOpen: boolean;
+  glossaryName: string;
+  onCancel: () => void;
+  onOk: () => void;
+}
+
+const ExportGlossaryModal: FC<Props> = ({
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Glossary/ExportGlossaryModal/ExportGlossaryModal.tsx:28]: unit tests / cypress tests

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryV1.component.tsx [hunk 3]
```diff
@@ -266,6 +276,68 @@ const GlossaryV1 = ({
...
+                    <Col span={21}>
+                      <Typography.Text
+                        className="font-medium"
+                        data-testid="import-button-title">
+                        {t('label.import')}
+                      </Typography.Text>
+                    </Col>
+                    <Col className="p-t-xss">
+                      <Typography.Paragraph className="text-grey-muted text-xs m-b-0 line-height-16">
+                        Import glossary terms
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryV1.component.tsx:330]: localisation

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/ImportResult/ImportResult.tsx [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+      {
+        title: t('label.tag-plural'),
+        dataIndex: 'tags',
+        key: 'tags',
+        render: (tags: GlossaryCSVRecord['tags']) => {
+          return <Typography.Text>{tags}</Typography.Text>;
+        },
+      },
+    ],
+    [parsedRecords]
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Glossary/ImportResult/ImportResult.tsx:118]: Not sure why we have deps for parsedRecords for columns data

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/ImportResult/ImportResult.tsx [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+          </div>
+        </Space>
+      </Col>
+      <Col span={24}>
+        <Table
+          bordered
+          columns={columns}
+          data-testid="import-result-table"
+          dataSource={parsedRecords}
+          pagination={false}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Glossary/ImportResult/ImportResult.tsx:159]: add `rowKey` props to the table

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/Glossary.test.tsx [hunk 6]
```diff
@@ -78,14 +78,12 @@ jest.mock('react-router-dom', () => ({
   useParams: jest.fn().mockReturnValue({
     glossaryName: 'GlossaryName',
   }),
+  Link: jest.fn().mockImplementation(({ children }) => <a>{children}</a>),
 }));
 
 jest.mock('components/GlossaryDetails/GlossaryDetails.component', () => {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Glossary/Glossary.test.tsx:84]: Can we rename this file to GlossaryV1?

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/ImportResult/ImportResult.tsx (hunk 4 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/Glossary/ImportResult/ImportResult.tsx:118]: We don't need that, I will remove it.

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/ImportGlossary/ImportGlossary.tsx [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+  const [csvImportResult, setCsvImportResult] = useState<CSVImportResult>();
+
+  const breadcrumbList: TitleBreadcrumbProps['titleLinks'] = useMemo(
+    () => [
+      {
+        name: glossaryName,
+        url: getGlossaryPath(glossaryName),
+      },
+      {
+        name: 'Import Glossary Terms',
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/Glossary/ImportGlossary/ImportGlossary.tsx:71]: localization

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/ImportGlossary/ImportGlossary.tsx [hunk 8]
```diff
@@ -0,0 +1,271 @@
...
+    useMemo(() => {
+      const isSuccess =
+        csvImportResult?.status === Status.Success ||
+        csvImportResult?.status === Status.PartialSuccess;
+      const isFailure = csvImportResult?.status === Status.Failure;
+      const showAbortedResult = csvImportResult?.status === Status.Aborted;
+      const showSuccessResult =
+        csvImportResult?.status === Status.Success ||
+        csvImportResult?.status === Status.PartialSuccess ||
+        csvImportResult?.status === Status.Failure;
```
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/src/components/Glossary/ImportGlossary/ImportGlossary.tsx:89]:
```suggestion
      const showSuccessResult = isSuccess || isFailure;
```

**Review by ShaileshParmar11 — APPROVED** [review]
miner comments but not blocker

**Pull request merged**